extensions/npm/src/commands.ts

```diff
@@ -28,4 +32,29 @@ export function runSelectedScript() {
 		let message = localize('noScriptFound', 'Could not find a valid npm script at the selection.');
 		vscode.window.showErrorMessage(message);
 	}
-}
\ No newline at end of file
+}
+
+export async function selectAndRunScriptFromFolder(folderInfo: vscode.Uri) {
+	type TaskMap = { [id: string]: vscode.Task; };
+	let taskList = await detectNpmScriptsForFolder(folderInfo.path);
+
+	if (taskList && taskList.length > 0) {
+		let taskMap: TaskMap = {};
+		taskList.forEach(t => {
+			let uri = getPackageJsonUriFromTask(t);
+			if (uri && uri.fsPath.length >= folderInfo.fsPath.length) {
+				let taskName = uri.fsPath.substring(folderInfo.fsPath.length, uri.fsPath.length - '/package.json'.length) + ' > ' + t.name.substring(0, t.name.search('-'));
```

Comment:
Using '>' to encode context information doesn't fit the existing UI and users will not understand it, since it is used nowhere else. The location of the script should be indicated in the same way as in 'Run Tasks', that is, it is rendered as a suffix and not encoded.

![image](https://user-images.githubusercontent.com/172399/66517825-21155500-eae4-11e9-9351-bbf65f89852e.png)